File: st-util/src/main/java/de/saring/util/unitcalc/CalculationUtils.java
package de.saring.util.unitcalc;

/** 
 * This class contains several static methods for calculating exercise data.
 * 
 * @author  Stefan Saring
 * @version 1.0
 */
public class CalculationUtils
{
    /**
     * Calculates the distance for the given AVG speed and duration.
     * 
     * @param avgSpeed avarage speed (km/h)
     * @param duration duration (seconds)
     * @return the distance (km)
     */
    public static float calculateDistance (float avgSpeed, int duration) {
        return (duration / 3600f) * avgSpeed;
    }
    
    /**
     * Calculates the AVG speed for the given distance and duration.
     * 
     * @param distance distance (km)
     * @param duration duration (seconds)
     * @return the AVG speed (km/h)
     */
    public static float calculateAvgSpeed (float distance, int duration) {
        return distance / (duration / 3600f);
    }
    
    /**
     * Calculates the duration for the given distance and AVG speed.
     * 
     * @param distance distance (km)
     * @param avgSpeed avarage speed (km/h)
     * @return the duration (seconds)
     */
    public static int calculateDuration (float distance, float avgSpeed) {
        return (int) ((distance / avgSpeed) * 3600);
    }
}

File: st-util/src/main/kotlin/de/saring/util/unitcalc/CalculationUtils.kt
package de.saring.util.unitcalc

/**
 * This class contains several static methods for calculating exercise data.
 *
 * @author Stefan Saring
 */
object CalculationUtils {

    /**
     * Calculates the distance for the given AVG speed and duration.
     *
     * @param avgSpeed average speed (km/h)
     * @param duration duration (seconds)
     * @return the distance (km)
     */
    @JvmStatic
    fun calculateDistance(avgSpeed: Double, duration: Int): Double =
            duration / 3600.0 * avgSpeed

    /**
     * Calculates the AVG speed for the given distance and duration.
     *
     * @param distance distance (km)
     * @param duration duration (seconds)
     * @return the AVG speed (km/h)
     */
    @JvmStatic
    fun calculateAvgSpeed(distance: Double, duration: Int): Double =
            distance / (duration / 3600.0)

    /**
     * Calculates the duration for the given distance and AVG speed.
     *
     * @param distance distance (km)
     * @param avgSpeed average speed (km/h)
     * @return the duration (seconds)
     */
    @JvmStatic
    fun calculateDuration(distance: Double, avgSpeed: Double): Int =
            (distance / avgSpeed * 3600).toInt()
}
